integration/disk_flush_helpers.go

```diff
@@ -45,7 +46,36 @@ var (
 )
 
 // nolint: deadcode
-func waitUntilDataFlushed(
+func waitUntilSnapshotFilesFlushed(
```

Comment:
nit: could you refactor both `waitUntilSnapshotFilesFlushed` method and `waitUntilDataFilesFlushed` to have a common function where you can pass in the function to resolve file name

Reply:
Yeah, we should probably bite the bullet and do that.

Reply:
They have different signatures :/

I can probably wrap the one that returns an error in another function that panics on the error though....I'll make it work

Reply:
done

Reply:
Ok I spent 15 minutes trying to do this and its not gonna be pretty. The function to check if they have a file on disk have different signatures, but the outer functions themselves also accept different arguments ([]time.Time vs. map[xtime.UnixNano]generate.SeriesBlock

I don't think this is a good refactor. There functions look superficially similar, but the differences are big enough that if I combine them into a shared function its going to be painfully DRY and more confusing than anything.

Gonna pass on refactoring this unless one of you feels really strongly that you have a good solution

Reply:
Np, I'm fine with that.

Reply:
Yeah def not worth the hassle